File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:

    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        m = max(piles)
        l,r=1,m
        res =r
        while l<=r:
            mid = l+(r-l)//2
            currT=0
            for pile in piles:
                currT += math.ceil(float(pile) / mid)
            if currT<=h:
                res=min(res, mid)
                r= mid-1
            else:
                l=mid+1
        return res

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-2.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        vector<int> lis(nums.size(),1);
        int res=1;
        for (int i=nums.size()-1;i>=0;i--){
            for(int j=i+1;j<nums.size();j++){
                if (nums[i]<nums[j]){
                    lis[i]=max(lis[j]+1,lis[i]);
                    res=max(res,lis[i]);
                }
            }
        }
            return res;

    }
};
